extensions/spark/kyuubi-spark-lineage/src/test/scala/org/apache/kyuubi/plugin/lineage/helper/SparkSQLLineageParserHelperSuite.scala

```diff
@@ -783,6 +785,29 @@ class SparkSQLLineageParserHelperSuite extends KyuubiFunSuite
 
   }
 
+  test("colums lineage extract - catch table") {
```

Comment:
The test case also seems to run successfully before this fix.

![image](https://user-images.githubusercontent.com/17894939/199865897-06b95d11-33b1-4bda-b303-389aa7bd9a64.png)

Reply:
We may need to add a test case in `OperationLineageEventSuite`.

Reply:
Yes, but we can not meet the real situation in the test code. In the `SparkOperationLineageQueryExecutionListener`,  the optimized plan from `QueryExecution.optimizedPlan` which will be optimized with `withCachedData` , so the execution plan is different from calling` spark.sessionState.optimizer.execut(analyzed)`, will be like this: 

```
Project [a#102 AS aa#83, b0#98 AS bb#84]
+- Join Inner, (a#102 = a0#97)
   :- Filter isnotnull(a0#97)
   :  +- InMemoryRelation [a0#97, b0#98], StorageLevel(disk, memory, deserialized, 1 replicas)
   :        +- *(1) Project [a#11 AS a0#9, b#12 AS b0#10]
   :           +- *(1) Filter (isnotnull(a#11) AND (a#11 = 1))
   :              +- Scan hive default.table0 [a#11, b#12], HiveTableRelation [`default`.`table0`, org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe, Data Cols: [a#11, b#12, c#13], Partition Cols: []]
   +- Project [a#102]
      +- Filter isnotnull(a#102)
         +- HiveTableRelation [`default`.`table1`, org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe, Data Cols: [a#102, b#103, c#104], Partition Cols: []]
```

Reply:
Added a complete test case for `cache table`. cc @wForget